File: 00-code(源代码)/src/com/hi/dhl/algorithms/offer/_10_2/java/Solution.java
package com.hi.dhl.algorithms.offer._10_2.java;

/**
 * <pre>
 *     author: dhl
 *     desc  :
 * </pre>
 */
class Solution {
    public int numWays(int n) {
        if (n == 0 || n == 1) {
            return 1;
        }

        int a = 1;
        int b = 2;
        int sum = b;
        for (int i = 2; i < n; i++) {
            sum = (a + b) % 1000000007;
            a = b;
            b = sum;
        }
        return sum;
    }

    public static void main(String... args) {
        Solution solution = new Solution();
        System.out.println(solution.numWays(7));
    }
}

File: 00-code(源代码)/src/com/hi/dhl/algorithms/offer/_10_2/kotlin/Solution.kt
package com.hi.dhl.algorithms.offer._10_2.kotlin

/**
 * <pre>
 *     author: dhl
 *     desc  :
 * </pre>
 */
class Solution {
    fun numWays(n: Int): Int {
        if (n == 1 || n == 0) {
            return 1;
        }

        var a = 1;
        var b = 2;
        var sum = b;
        for (i in 2 until n) {
            sum = (a + b) % 1000000007
            a = b.also { b = sum }
        }
        return sum
    }
}

fun main(args: Array<String>) {
    val solution = Solution()
    println(solution.numWays(100))//782204094
}